### DISCUSSION TITLE
Proper start-values for some of the following iteration variables might help

### DISCUSSION THREAD

**Discussion by bflo96** [OpenModelica/OpenModelica#13244] [General]
Hello everyone, 

I am using OpenModelica to simulate a circuit with a significant number of resistors in parallel and series. They're connected to a current source with a known current value.

The resistance of each resistor is dependent on the current that goes through it: v = R(i) * i. 
I am aware my problem has a significant non-linearity due to the way I am modeling the resistances' values.

When I try to simulate, it gives me this error:

> nonlinear system 461 fails: at t=0
> Model_functions.c:696: Invalid root: (0)^(-0.909096)

> proper start-values for some of the following iteration variables might help
> [1] Real r[15].i(start=1, nominal=1)
> [2] Real r[14].i(start=1, nominal=1)
> [3] Real r[13].i(start=1, nominal=1)
> [4] Real r[12].i(start=1, nominal=1)
> [5] Real r[11].i(start=1, nominal=1)
> [6] Real r[10].i(start=1, nominal=1)
> [7] Real r[9].i(start=1, nominal=1)
> [8] Real r[8].i(start=1, nominal=1)
> etc

It doesn't work no matter the start value of the current I choose, because the root will always fall at zero at t = 0.

I also tried to set up initial equations, but they're considered redundant.

Is there any way I can troubleshoot what is causing this problem?
Can the problem be too non-linear and not solvable?

Thank you in advance.

**Comment by casella** [comment]
You can check with the [Equation-Based Debugger](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/debugger.html#the-equation-based-debugger) how the model is solved. In case of nonlinear systems of equations, you can see what are the iteration variables for the Newton method and their start attributes.

**Reply by bflo96** [reply]
I eventually tried Transformational Debugger instead, which I think it shows the start values as well.
I made some changes in my model and is now showing a division by zero.

I know the following variable is causing the error , but even if I try to define a fixed start value for different from zero, it will still show up like below:
> (assign) Modelo.flow[1,1] := 0

What can I possibly be doing wrong? 

EDIT: I was wondering if there's any way I can say that a certain variable can never be zero.

**Reply by casella** [reply]
You can give that a small min attribute, e.g.
```modelica
Real x(min = 1e-9);
```

**Reply by bflo96** [reply]
I tried this approach as well. 
But it's not valid for my problem because the variable I was trying to restrict can actually be a negative number. It just can never be zero. 

However, I managed to find the error - it was on another part of the code, namely when creating an array.
Your suggestion to look into the start values was really helpful! I eventually managed to figure out the initial start values weren't making any sense regarding what I intended to simulate 😊
I have another problem now, but I think it's probably better to create a different discussion for that.

Thank you so much!

**Comment by casella** [comment]
Could you post your example in a ticket?

**Reply by bflo96** [reply]
Hi @casella! I managed to solve this error (problem on my side identified through your suggestion)
I have another problem now, because the program just crashes when I increase the size of the system. 

Should I create a different discussion for this? 

```
Process crashed
Simulation process failed. Exited with code 0xffffffffc0000005.
```

**Reply by casella** [reply]
Please do. If you can share the code so we can reproduce the issue, please open a proper issue an post it there.
